datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -432,6 +432,27 @@ impl Interval {
         }
     }
 
+    /// Compute the logical disjunction of this (boolean) interval with the given boolean interval.
+    pub(crate) fn or<T: Borrow<Interval>>(&self, other: T) -> Result<Interval> {
+        let rhs = other.borrow();
+        if self.get_datatype()? != DataType::Boolean
+            && rhs.get_datatype()? != DataType::Boolean
```

Comment:
```suggestion
        if self.get_datatype()? != DataType::Boolean
            || rhs.get_datatype()? != DataType::Boolean
```

I am not sure if different data types can `OR` operation.

Reply:
Good call -- thank you @Weijun-H